Support Python 3.14t

pquentin opened the pull request:
Now that cffi and cryptography support free-threading, it's possible to test it in CI.

pquentin commented:
The lint failures are caused by the dependencies bump. I didn’t investigate yet.

illia-v commented on .github/workflows/ci.yml:
```diff
@@ -146,13 +146,6 @@ jobs:
-        if: runner.os == 'Windows'
-        shell: cmd
-        run: |
-          mklink "%Python3_ROOT_DIR%\python${{ matrix.python-version }}.exe" "%Python3_ROOT_DIR%\python.exe"
```
It was fixed by https://github.com/wntrblm/nox/pull/989, right?

pquentin commented:
> They are caused by the mypy upgrade most likely.

Indeed! I reverted the mypy and h2 upgrades.

> I've noticed a weird mismatch in Python versions used for Ubuntu 3.14t and Windows 3.14t runs:

Good catch! I fixed this by upgrading uv from 0.7 to 0.9. The uv version is a parameter of the setup-uv action, and we don't have any automation to upgrade it. I also disabled memray for free-threading builds as it currently crashes.

Please take another look!

kuraga commented:
`3.13t`?
